### PR TITLE
doc: update must done during settlement

### PR DESCRIPTION
**Pull request by hackaugusto**

### CODE REVIEW

File: raiden_contracts/contracts/TokenNetwork.sol [hunk 1]
```diff
@@ -558,7 +558,27 @@ contract TokenNetwork is Utils {
...
+        //   balance proofs which should be provided by A is missing, so B will
+        //   call settle with its balance proof zeroed out.
+        // 3 A is restarts and calls update, which will change B's balance
+        //   proof.
+        // 4 At this point, the transactions from 2 and 3 are racing, and one
+        //   of them will fail.
+        //
+        // To avoid the above race condition, which would require special
+        // handling on both nodes, the call to update is forbidden after the
+        // settlement window. This does not affect safity, since we assume the
```
Comment (rakanalh) [inline raiden_contracts/contracts/TokenNetwork.sol]: `s/safity/safety`

File: raiden_contracts/contracts/TokenNetwork.sol [hunk 2]
```diff
@@ -558,7 +558,27 @@ contract TokenNetwork is Utils {
...
+        // nodes are always properly configure and have a monitoring service
+        // available to cal update on the user's behalf.
```
Comment (rakanalh) [inline raiden_contracts/contracts/TokenNetwork.sol]: `s/cal/call`

File: raiden_contracts/contracts/TokenNetwork.sol [hunk 3]
```diff
@@ -558,7 +558,27 @@ contract TokenNetwork is Utils {
...
 
         require(channel.state == ChannelState.Closed);
-        // Channel must be in the settlement window
+
+        // Calling this function after the settlement window is forbidden to
+        // fix the following race condition:
+        //
+        // 1 A badly configured node A, that doesn't have a monitoring service
+        //   and is temporarily offline does not call update during the
+        //   settlement window
```
Comment (rakanalh) [inline raiden_contracts/contracts/TokenNetwork.sol]: Nitpick: end with a full stop.

File: raiden_contracts/contracts/TokenNetwork.sol [hunk 4]
```diff
@@ -558,7 +558,27 @@ contract TokenNetwork is Utils {
...
+        // 2 The well behaved partner B, which called close, sees the
```
Comment (rakanalh) [inline raiden_contracts/contracts/TokenNetwork.sol]: `s/which/who`

File: raiden_contracts/contracts/TokenNetwork.sol [hunk 5]
```diff
@@ -558,7 +558,27 @@ contract TokenNetwork is Utils {
...
+        // fix the following race condition:
+        //
+        // 1 A badly configured node A, that doesn't have a monitoring service
+        //   and is temporarily offline does not call update during the
+        //   settlement window.
+        // 2 The well behaved partner B, who called close, sees the
+        //   settlement window is over and calls settle. At this point the B's
+        //   balance proofs which should be provided by A is missing, so B will
+        //   call settle with its balance proof zeroed out.
+        // 3 A is restarts and calls update, which will change B's balance
```
Comment (LefterisJP) [inline raiden_contracts/contracts/TokenNetwork.sol]: A is restarts -> A restarts

File: raiden_contracts/contracts/TokenNetwork.sol [hunk 6]
```diff
@@ -558,7 +558,27 @@ contract TokenNetwork is Utils {
...
+        //   proof.
+        // 4 At this point, the transactions from 2 and 3 are racing, and one
+        //   of them will fail.
+        //
+        // To avoid the above race condition, which would require special
+        // handling on both nodes, the call to update is forbidden after the
+        // settlement window. This does not affect safety, since we assume the
+        // nodes are always properly configure and have a monitoring service
```
Comment (LefterisJP) [inline raiden_contracts/contracts/TokenNetwork.sol]: configure -> configured

File: raiden_contracts/contracts/TokenNetwork.sol [hunk 7]
```diff
@@ -558,7 +558,27 @@ contract TokenNetwork is Utils {
         Channel storage channel = channels[channel_identifier];
 
         require(channel.state == ChannelState.Closed);
-        // Channel must be in the settlement window
+
+        // Calling this function after the settlement window is forbidden to
```
Comment (LefterisJP) [inline raiden_contracts/contracts/TokenNetwork.sol]: As discussed in rocketchat contract changes, even comments, should not go in before Red Eyes. So I am just adding a blocking review until then.

**Pull request merged**